pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java

```diff
@@ -262,13 +264,24 @@ private IndexedTable getIndexedTable(DataSchema dataSchema, Collection<DataTable
     int numDataTables = dataTablesToReduce.size();
 
     // Get the number of threads to use for reducing.
-    int numReduceThreadsToUse = getNumReduceThreadsToUse(numDataTables, reducerContext.getMaxReduceThreadsPerQuery());
-
     // In case of single reduce thread, fall back to SimpleIndexedTable to avoid redundant locking/unlocking calls.
-    int capacity = GroupByUtils.getTableCapacity(_queryContext);
-    IndexedTable indexedTable =
-        (numReduceThreadsToUse > 1) ? new ConcurrentIndexedTable(dataSchema, _queryContext, capacity)
-            : new SimpleIndexedTable(dataSchema, _queryContext, capacity);
+    int numReduceThreadsToUse = getNumReduceThreadsToUse(numDataTables, reducerContext.getMaxReduceThreadsPerQuery());
+    int trimSize = GroupByUtils.getTableCapacity(_queryContext);
+    int trimThreshold = reducerContext.getSqlGroupByTrimThreshold();
+    IndexedTable indexedTable;
+    if (numReduceThreadsToUse <= 1) {
+      indexedTable = new SimpleIndexedTable(dataSchema, _queryContext, trimSize, trimThreshold);
+    } else {
+      if (trimThreshold == GroupByOrderByCombineOperator.INFINITE_INTER_SEGMENT_GROUPS_LIMIT) {
```

Comment:
Use `>=` instead of `==`?

Reply:
+1

Reply:
Right. I did the same in server combine. Missed it here. Fixed now